### PR TITLE
V9 : Port Currency Rate update

### PR DESCRIPTION
**Pull request by vrenaville**
- Remove check on base=True main currency, used company currency instead

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This migration should be taken as opportunity to refactor this bloated module to isolate update engine from each of the possible update sources. Maybe one or 2 of the more generals sources can be inside the main module, but not all of them.

File: currency_rate_update/__openerp__.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 ##############################################################################
 {
     "name": "Currency Rate Update",
-    "version": "8.0.0.7.0",
+    "version": "9.0.0.7.0",
```
Comment (elicoidal) [inline currency_rate_update/__openerp__.py]: 9.0.1.0.0

File: currency_rate_update/model/company.py [hunk 2]
```diff
@@ -25,15 +25,6 @@
...
-        "check if multi company currency is enabled"
-        company_currency = self.env['res.currency'].search([('company_id',
-                                                             '!=', False)])
-        for company in self:
-            company.multi_company_currency_enable = \
-                1 if company_currency else 0
-
     @api.one
     def button_refresh_currency(self):
         """Refresh the currencies rates !!for all companies now"""
```
Comment (elicoidal) [inline currency_rate_update/model/company.py]: Refresh the currencies rates (for all companies now).

**Comment by okuryan** [comment]
What is status of this pull request? We need this module for our customer and would like to decide either port to v9 ourselves or use existant module

**Comment by pedrobaeza** [comment]
As I said before, this should be refactored to have a base with base functionality and a global source, and then several submodules depending on it with the possible sources.

**Comment by jgrandguillaume** [comment]
Hi,

We don't have the resources yet to refactor this module properly (very old one I agree). But still, lot's of people need it on v9.0. Can we accept it like this ? It is not really well designed and I agree with Pedro, but on other side it works... 

Unless someone raise a hand to refactor it, I'm giving my :+1:  to merge it once the little nits raised by Eric has been cleaned up.

Regards,

Joël

**Comment by yvaucher** [comment]
This PR is focused to make admin ch currency update working. We don't intend to do the other as we lack of resource to do it. This could be a base for other, maybe we should raise a not ported warning for other types of currency import ?

:+1: as soon as comments from @elicoidal  are fixed

**Comment by vrenaville** [comment]
@yvaucher Finally I fixed reported issues

**Comment by yvaucher** [comment]
@katyukha Thanks for your work, I included it here

**Comment by ghost** [comment]
Hello! What is pending in this pull request? May I help?

**Comment by yvaucher** [comment]
@gs-clearcorp I think it was only missing reviews. We can now merge it.

**Pull request merged**